### PR TITLE
Client (Python): Msgpack implementation

### PR DESCRIPTION
**Pull request by cbrzn**

### CODE REVIEW

File: packages/py/msgpack/msgpack_wrap/__init__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import msgpack
+from dataclasses import dataclass, is_dataclass
+
+
+def sanitize(v):
+    for key in v.keys():
+        if is_dataclass(v[key]):
+            v[key] = v[key].__dict__
```
Comment (Niraj-Kamdar) [inline packages/py/msgpack/msgpack_wrap/__init__.py]: do not access `__dict__` directly.

use `asdict` function from the dataclassses package.

File: packages/py/msgpack/tests/test_msgpack.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    custom_object = {
+        "firstKey": "firstValue",
+        "secondKey": "secondValue"
+    }
+
+    encoded = msgpack_encode(custom_object)
+    assert encoded == bytes(expected_array_like)
+
+    decoded = msgpack_decode(encoded)
+    assert decoded == custom_object
```
Comment (Niraj-Kamdar) [inline packages/py/msgpack/tests/test_msgpack.py]: encode dictionary as our generic map and encode dataclasses as the object (msgpack map family).

**Review by Niraj-Kamdar — CHANGES_REQUESTED** [review]

File: packages/py/msgpack/msgpack_wrap/__init__.py (hunk 1 above)
Comment (Niraj-Kamdar) [reply packages/py/msgpack/msgpack_wrap/__init__.py]: You don't need to recurse when using asdict, it will take care of it for you

Comment (cbrzn) [reply packages/py/msgpack/msgpack_wrap/__init__.py]: I think we still want to have recursion here. The reason is that if the value is a dictionary and has nested instances of classes, we need to check those. Does this make sense to you?

Comment (Niraj-Kamdar) [reply packages/py/msgpack/msgpack_wrap/__init__.py]: Gotcha! It makes sense in that case but try to run this code in python3.10 and you will understand why it is bad idea to access `__dict__` :)

```python
from dataclasses import dataclass

@dataclass
class Test1:
    a: str
    
@dataclass(slots=True)
class Test2:
    a: str
    
print(Test1(a="1"))
print(Test1(a="1").__dict__)

print(Test2(a="2"))
print(Test2(a="2").__dict__)

```

Comment (cbrzn) [reply packages/py/msgpack/msgpack_wrap/__init__.py]: Yah I removed the `__dict__` and we are now using `asdict` :smile:

**Pull request merged**